File: programmers/level2/가장_큰_수.java
package programmers;
import java.util.*;
public class 가장_큰_수 {
	static
	class Solution {
	    class Node implements Comparable<Node>{
	    	String num;
	    	Node(String num){
	    		this.num = num;
	    	}
			@Override
			public int compareTo(Node o) {
				if(Integer.parseInt(this.num+o.num) > Integer.parseInt(o.num+this.num)) {
					return -1;
				}
				if(Integer.parseInt(this.num+o.num) < Integer.parseInt(o.num+this.num)) {
					return 1;
				}
				return 0;
			}
	    }
		public String solution(int[] numbers) {
	        StringBuilder answer = new StringBuilder();
	        Node[] arr = new Node[numbers.length];
	        for(int i = 0; i < numbers.length; ++i) {
	        	arr[i] = new Node(String.valueOf(numbers[i]));
	        }
	        Arrays.sort(arr);
	        for(Node n : arr) {
	        	answer.append(n.num);
	        }
	        if(answer.charAt(0)=='0'){
	            answer = new StringBuilder("0");
	        }
	        return answer.toString();
	    }
	}
	public static void main(String[] args) {
		int[] numbers = {0,0,0,0};
		Solution sol = new Solution();
		System.out.println(sol.solution(numbers));
	}
}

File: programmers/python/level2/가장_큰_수.py
class Obj:
    def __init__(self, num):
        self.num = num
    def __lt__(self, other):
        sum1 = int(str(self.num) + str(other.num))
        sum2 = int(str(other.num) + str(self.num))
        if sum1 > sum2:
            return True
        else:
            return False

def solution(numbers):
    arr = [Obj(num) for num in numbers]
    arr.sort()
    return ''.join(map(lambda e: str(e.num) , arr))

print(solution([0,0]))